src/superpmi-collect/superpmicollect.cs

```diff
@@ -0,0 +1,777 @@
+            {
+               string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+               arguments = program;
+               program = "dotnet";
+            }
+            
+            // If the program is a script, move the program name into the arguments, and run it
+            // under the appropriate shell.
+            if (Global.IsWindows)
+            {
+                if ((program.LastIndexOf(".bat") != -1) || (program.LastIndexOf(".cmd") != -1))
+                {
+                    string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+                    arguments = "/c " + program + programArgumentSep + arguments;
+                    program = Environment.GetEnvironmentVariable("ComSpec"); // path to CMD.exe
+                }
+            }
+            else
+            {
+                if (program.LastIndexOf(".sh") != -1)
+                {
+                    string programArgumentSep = String.IsNullOrEmpty(arguments) ? "" : " ";
+                    arguments = "bash " + program + programArgumentSep + arguments;
+                    program = "/usr/bin/env";
+                }
+            }
+
+            Console.WriteLine("Running: " + program + " " + arguments);
+            Process p = Process.Start(program, arguments);
+            p.WaitForExit();
+            return p.ExitCode;
+        }
+
+        // Run a single test from the coreclr test binary drop.
+        // This works even if given a test path in Windows file system format (e.g.,
+        // "c:\foo\bar\runit.cmd") when run on Unix. It converts to Unix path format and replaces
+        // the ".cmd" with ".sh" before attempting to run the script.
+        private static void RunTest(string testName)
+        {
+            string testDir;
+
+            if (Global.IsWindows)
+            {
+                int lastIndex = testName.LastIndexOf("\\");
+                if (lastIndex == -1)
+                {
+                    throw new SpmiException("test path doesn't have any directory separators? " + testName);
+                }
+                testDir = testName.Substring(0, lastIndex);
+            }
+            else
+            {
+                // Just in case we've been given a test name in Windows format, convert it to Unix format here.
+
+                testName = testName.Replace("\\", "/");
+                testName = testName.Replace(".cmd", ".sh");
+                testName = testName.Replace(".bat", ".sh");
+
+                // The way tests are run on Linux, we might need to do some setup. In particular,
+                // if the test scripts are copied from Windows, we need to convert line endings
+                // to Unix line endings, and make the script executable. We can always do this
+                // more than once. This same transformation is done in runtest.sh.
+                // Review: RunProgram doesn't seem to work if the program isn't a full path.
```

Comment:
Maybe move the check to make sure we have a absolute path before the RunProgram calls?